Plugins/NativeCamera/Source/Android/NativeCameraImpl.cpp

```diff
@@ -166,10 +242,40 @@ namespace Babylon::Plugins
             m_deviceContext = &Graphics::DeviceContext::GetFromJavaScript(m_env);
         }
 
+        if (maxWidth == 0 || maxWidth > std::numeric_limits<int32_t>::max()) {
+            maxWidth = std::numeric_limits<int32_t>::max();
+        }
+        if (maxHeight == 0 || maxHeight > std::numeric_limits<int32_t>::max()) {
+            maxHeight = std::numeric_limits<int32_t>::max();
+        }
+
         return android::Permissions::CheckCameraPermissionAsync().then(arcana::inline_scheduler, arcana::cancellation::none(), [this, maxWidth, maxHeight, frontCamera]()
         {
-            m_cameraDimensions.width = maxWidth;
-            m_cameraDimensions.height = maxHeight;
+            // Get the phone's current rotation so we can determine if the camera image needs to be rotated based on the sensor's natural orientation
+            int phoneRotation{ GetAppContext().getSystemService<android::view::WindowManager>().getDefaultDisplay().getRotation() * 90 };
+            CameraConfiguration bestCameraConfiguration{"", maxWidth, maxHeight, phoneRotation};
+
+            if (API_LEVEL >= 24 && libCamera2NDK && !m_overrideCameraTexture) {
+                m_cameraManager = GET_CAMERA_FUNCTION(ACameraManager_create)();
+                bestCameraConfiguration = GetCameraId(maxWidth, maxHeight, frontCamera);
+
+                // If no matching device, throw an error with the message "ConstraintError" which matches the behavior in the browser.
+                if (bestCameraConfiguration.cameraID.empty())
+                {
+                    throw std::runtime_error{"ConstraintError: Unable to match constraints to a supported camera configuration."};
+                }
+            }
+
+            // The sensor rotation dictates the orientation of the camera when the phone is in it's default orientation
+            // Subtracting the phone's rotation from the camera's rotation will give us the current orientation
+            // of the sensor. Then add 360 and modulus 360 to ensure we're always talking about positive degrees.
+            int sensorRotationDiff = (bestCameraConfiguration.sensorRotation - phoneRotation + 360) % 360;
+            bool sensorIsPortrait = sensorRotationDiff == 90 || sensorRotationDiff == 270;
```

Comment:
Uniform initialization for these and other vars:
```suggestion
            int sensorRotationDiff{(bestCameraConfiguration.sensorRotation - phoneRotation + 360) % 360};
            bool sensorIsPortrait{sensorRotationDiff == 90 || sensorRotationDiff == 270};
```